### ISSUE TITLE
custom-property-empty-line-before

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#1578]
- Name: `custom-property-empty-line-before`
- Primary option: `"always"|"never"`
- Secondary options: `{ "except": ["first-nested", "after-comment", "after-custom-property"] }`
- Messages: 
  - Rejected: "Unexpected empty line before custom property"
  - Expected: "Expected empty line before custom property"
- Section: "Custom property"

Given:

```js
[ "always", { "except": ["first-nested", "after-comment", "after-custom-property"] }]
```

The following is valid:

```css
a {
  --custom-prop: value; /* "always" except first-nested */
  --custom-prop2: value; /* "always" except after-custom-property */

  /* comment */
  --custom-prop3: value; /* "always" except after-comment */

  @extends "x";

  --custom-prop4: value; /* primary option is "always" */

   & b {
     prop: value;
   }

   --custom-prop5: value;  /* primary option is "always" */
}
```

Given:

```js
"never"
```

The following is valid:

```css
a {
  --custom-prop: value; /* primary option is "never" */
  --custom-prop2: value; /* primary option is "never" */

  /* comment */
  --custom-prop3: value; /* primary option is "never" */

  @extends "x";
  --custom-prop4: value; /* primary option is "never" */

   & b {
     prop: value;
   }
   --custom-prop5: value; /* primary option is "never" */
}
```

Ref: https://github.com/stylelint/stylelint/issues/1500#issuecomment-230086153

**Comment by sendilkumarn** [comment]
I will give this a try, if no one has taken this issue

**Comment by jeddy3** [comment]
> if no one has taken this issue

I believe it's up for grabs.

Be sure to read the [developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md). You'll find a decent blueprint in the [`rule-non-nested-empty-line-before`](https://github.com/stylelint/stylelint/tree/master/src/rules/rule-non-nested-empty-line-before) to help you get started. A recently accept new rule PR looks [like this](https://github.com/stylelint/stylelint/pull/1629/files).

As an aside, I suspect a few of us will be taking a little breather now that `7.0.0` is out as a lot of hard work went into that release. So, don't worry if the review process is a little slower than usual.
